h2o-algos/src/main/java/hex/gam/GAM.java

```diff
@@ -0,0 +1,144 @@
+        error("_link", "Only identity or family_default link is supported for now.");
+      if (_parms._gam_X==null)
+        error("_gam_X", "must specify columns indices to apply GAM to.  If you don't have any, use GLM.");
+      if (_parms._k==null) {  // user did not specify any knots, we will use default 10, evenly spread over whole range
+        int numKnots = _train.numRows() < 10?(int)_train.numRows():10;
+        _parms._k = new int[_parms._gam_X.length];  // different columns may have different 
+        Arrays.fill(_parms._k, numKnots);
+      }
+    }
+  }
+
+  @Override
+  public void checkDistributions() {  // will be called in ModelBuilder.java
+    if (!_response.isNumeric()) {
+      error("_response", "Expected a numerical response, but instead got response with " + _response.cardinality() + " categories.");
+    }
+  }
+
+  @Override
+  protected boolean computePriorClassDistribution() {
+    return false; // no use, we don't output probabilities
+  }
+
+  @Override
+  protected int init_getNClass() {
+    return 1; // only regression is supported for now
+  }
+
+  @Override
+  protected GAMDriver trainModelImpl() {
+    return new GAMDriver();
+  }
+  
+  @Override 
+  protected int nModelsInParallel(int folds) {
+    return nModelsInParallel(folds, 2);
+  }
+
+  @Override protected void checkMemoryFootPrint_impl() {
+    ;
+  }
+  private class GAMDriver extends Driver {
+    /***
+     * This method will take the _train that contains the predictor columns and response columns only and add to it
+     * the following:
+     * 1. For each predictor included in gam_x, expand it out to calculate the f(x) and attach to the frame.
+     * @return
+     */
+    Frame adaptTrain() {
+      Frame orig = _parms.train();  // contain all columns, _train contains only predictors and responses
+      int numGamFrame = _parms._gam_X.length;
+      Key<Frame>[] gamFramesKey = new Key[numGamFrame];  // store the Frame keys of generated GAM column
+      RecursiveAction[] generateGamColumn = new RecursiveAction[numGamFrame];
+      for (int index=0; index < numGamFrame; index++) {
+        final Frame predictVec = new Frame(orig.vec(_parms._gam_X[index]));  // extract the vector to work on
+        final int numKnots = _parms._k[index];  // grab number of knots to generate
+        final BSType splineType = _parms._bs[index];
+        final int tIndex = index;
+        final String[] newColNames = new String[numKnots];
+        for (int colIndex = 0; colIndex < numKnots; colIndex++) {
+          newColNames[colIndex] = predictVec.name(0)+"_"+splineType.toString()+"_"+colIndex;
+        }
+        generateGamColumn[tIndex] = new RecursiveAction() {
+          @Override
+          protected void compute() {
+            Frame oneAugmentedColumn = new GenerateGamMatrixOneColumn(splineType, numKnots, null, predictVec,
+                    _parms._standardize).doAll(numKnots,Vec.T_NUM,predictVec).outputFrame(Key.make(), newColNames, 
```

Comment:
Michalk:  I was able to call GenerateGamMatrixOneColumn here with no problem at all!